### PR TITLE
Feature/us17 second language

### PR DESCRIPTION
**Pull request by ufelgen**
following [this US](https://github.com/ufelgen/capstone-project/issues/30) I added an option to save a second translation for a word.

### CODE REVIEW

File: components/AddTranslationForm/AddTranslationForm.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    const fields = event.target.elements;
+    const language2 = fields.queryLanguage2.value.split("-");
+    const translation2 = fields.queryLanguage2Translation.value;
+    const gender2 = fields.gender2.value;
+
+    const query2 = {
+      language: language2[1],
+      flag: language2[0],
+      translation: translation2,
+      gender: gender2,
```
Comment (onebarloop) [inline components/AddTranslationForm/AddTranslationForm.js:21]: do you need the 2s in the variable names here? Looks a bit odd - then again, its just the looks...

File: components/AddTranslationForm/AddTranslationForm.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    };
+    onSaveTranslation(id, query2);
+    onReturnFromEditMode();
+  }
+
+  return (
+    <StyledForm onSubmit={handleAddedTranslation}>
+      <label htmlFor="queryLanguage1" name="queryLanguage1">
+        <Dropdown name="queryLanguage2" id="queryLanguage2" required>
```
Comment (onebarloop) [inline components/AddTranslationForm/AddTranslationForm.js:30]: If you wrap your input inside a label, you don't need "htmlFor" and "id".  The reference is done via the nesting. And I am like 80% sure, the label-tag does not support the name-attribute.

File: components/PopupMenu/PopupMenu.js [hunk 3]
```diff
@@ -56,6 +57,15 @@ export default function PopupMenu({
               {prop.notes ? "notes" : "+ notes"}
             </StyledMenuButton>
           </Link>
+          {prop.query2 ? null : (
```
Comment (onebarloop) [inline components/PopupMenu/PopupMenu.js:60]: you don't need a ternary operator here, you could check if the opposite of your condition is true: 

```javascript
{!prop.query && (...) }
```

File: components/VocabCard/VocabCard.js [hunk 4]
```diff
@@ -22,18 +27,35 @@ export default function VocabCard({
...
           onClosePopup={onClosePopup}
           id={word.id}
           onDelete={onDelete}
           onEdit={onEdit}
+          onAddTranslation={(event) => onAddTranslation(event, word.id)}
           prop={word}
         />
-      ) : (
+      )}
+      {word.id === addTranslation ? null : (
```
Comment (onebarloop) [inline components/VocabCard/VocabCard.js:58]: See my comment above

**Review by onebarloop — APPROVED** [review]
This is intense Uli! I really this logic. 

Left some ideas, but nothing critical.

File: components/AddTranslationForm/AddTranslationForm.js (hunk 1 above)
Comment (ufelgen) [reply components/AddTranslationForm/AddTranslationForm.js:21]: I need to distinguish them from the values of query1... so I would say yes.. but you're right, it looks a bit strange ;)

File: components/AddTranslationForm/AddTranslationForm.js (hunk 2 above)
Comment (ufelgen) [reply components/AddTranslationForm/AddTranslationForm.js:30]: Thanks! I'll look into it!

File: components/PopupMenu/PopupMenu.js (hunk 3 above)
Comment (ufelgen) [reply components/PopupMenu/PopupMenu.js:60]: yes!! I always forget about the ! 🙈

File: components/AddTranslationForm/AddTranslationForm.js (hunk 1 above)
Comment (onebarloop) [reply components/AddTranslationForm/AddTranslationForm.js:21]: The variables are local to your function (you define them only here), there should be no conflict. 

Of course the name prop after "fields." must be the same as in your jsx

**Pull request merged**